src/interfaces/chain.h

```diff
@@ -62,6 +62,12 @@ class FoundBlock
     //! Read block data from disk. If the block exists but doesn't have data
     //! (for example due to pruning), the CBlock variable will be set to null.
     FoundBlock& data(CBlock& data) { m_data = &data; return *this; }
+    //! Read block's undo data from disk.
```

Comment:
This commit is quite long, making it difficult to focus on the main thing which is the scanning logic. I'd suggest breaking out smaller commits for just the mempool related changes, and just the undoblock related changes. Possibly more, but will need to dig into the commit a bit more.

Reply:
I broke this into 3 commits.